### PR TITLE
dotCMS/core#22140 Contentlet loosing in the page after layout change

### PR DESCRIPTION
**Pull request by rjvelazco**
**See how it works after the changes [here](https://www.loom.com/share/f180b405c1a84f3dbb8bc44a6ce6e490)**

### CODE REVIEW

File: apps/dotcms-ui/src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.spec.ts [hunk 1]
```diff
@@ -204,6 +204,25 @@ describe('DotEditLayoutComponent', () => {
...
+
+            // Trigger the observable
+            layoutDesignerDe.triggerEventHandler('updateTemplate', fakeLayout);
+
+            // Destroy the observable
+            component.destroy$.next(true);
+            component.destroy$.complete();
+            tick(10000);
+
+            // expect(dotPageLayoutService.save).toHaveBeenCalled();
```
Comment (hmoreras) [inline apps/dotcms-ui/src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.spec.ts:222]: remove

**Review by hmoreras — APPROVED** [review]

**Comment by alfredo-dotcms** [comment]
@rjvelazco for some reason only 3 test were run

image

**Comment by rjvelazco** [comment]
> @rjvelazco for some reason only 3 test were run
> 
> image

**Review by alfredo-dotcms — APPROVED** [review]

File: apps/dotcms-ui/src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.ts [hunk 2]
```diff
@@ -122,25 +123,28 @@ export class DotEditLayoutComponent implements OnInit, OnDestroy {
...
+                        .save(this.pageState.page.identifier, {
+                            ...layout,
+                            title: null
+                        })
+                        .pipe(finalize(() => this.canRouteBeDesativated(true)));
+                }),
+                tap({
+                    next: (updatedPage: DotPageRender) =>
+                        this.handleSuccessSaveTemplate(updatedPage),
+                    error: (err: ResponseView) => this.handleErrorSaveTemplate(err)
```
Comment (fmontes) [inline apps/dotcms-ui/src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.ts:144]: Shouldn't this be in the subscribe?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: apps/dotcms-ui/src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.ts (hunk 2 above)
Comment (rjvelazco) [reply apps/dotcms-ui/src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.ts:144]: Right!

**Review by oidacra — APPROVED** [review]

**Review by alfredo-dotcms — APPROVED** [review]

**Comment by rjvelazco** [comment]
> @rjvelazco for some reason only 3 test were run
> 
> image

@alfredo-dotcms It looks like only those 3 tests have been running since this change was merged but, after the change, all tests passed! 😄

https://github.com/dotCMS/core-web/blob/b467060ed3e94d9e04c2559a5ce33ce91ff0d3f2/apps/dotcms-ui/src/app/view/components/_common/dot-inline-edit/dot-inline-edit.component.spec.ts#L31-L74

**Review by fmontes — APPROVED** [review]

**Pull request merged**